<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title> publications | Vaibhav Mohanty, PhD </title> <meta name="author" content="Vaibhav Mohanty, PhD"> <meta name="description" content="Vaibhav Mohanty's academic webpage."> <meta name="keywords" content="vaibhav mohanty, fitness landscape design, physics of evolution, biophysics, statistical physics"> <link rel="stylesheet" href="/assets/css/bootstrap.min.css?a4b3f509e79c54a512b890d73235ef04"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link defer rel="stylesheet" href="/assets/css/academicons.min.css?f0b7046b84e425c55f3463ac249818f5"> <link defer rel="stylesheet" href="/assets/css/scholar-icons.css?62b2ac103a88034e6882a5be5f3e2772"> <link defer rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons&amp;display=swap"> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-github.css?591dab5a4e56573bf4ef7fd332894c99" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="data:image/svg+xml,&lt;svg%20xmlns=%22http://www.w3.org/2000/svg%22%20viewBox=%220%200%20100%20100%22&gt;&lt;text%20y=%22.9em%22%20font-size=%2290%22&gt;%E2%9B%B0%EF%B8%8F&lt;/text&gt;&lt;/svg&gt;"> <link rel="stylesheet" href="/assets/css/main.css?d41d8cd98f00b204e9800998ecf8427e"> <link rel="canonical" href="https://vaibhavmohanty.com/publications/"> <script src="/assets/js/theme.js?a81d82887dd692e91686b43de4542f18"></script> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-native.css?5847e5ed4a4568527aa6cfab446049ca" media="none" id="highlight_theme_dark"> <script>
    initTheme();
  </script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top" role="navigation"> <div class="container"> <a class="navbar-brand title font-weight-lighter" href="/"> <span class="font-weight-bold">Vaibhav</span> Mohanty, PhD </a> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item "> <a class="nav-link" href="/">about </a> </li> <li class="nav-item active"> <a class="nav-link" href="/publications/">publications <span class="sr-only">(current)</span> </a> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="ti ti-sun-moon" id="light-toggle-system"></i> <i class="ti ti-moon-filled" id="light-toggle-dark"></i> <i class="ti ti-sun-filled" id="light-toggle-light"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5" role="main"> <div class="post"> <header class="post-header"> <h1 class="post-title">publications</h1> <p class="post-description"></p> </header> <article> <div class="publications"> <h2 class="bibliography">2025</h2> <ol class="bibliography"> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <img src="/assets/img/publication_preview/fld_original.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="fld_original.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="mohanty_biophysical_2025" class="col-sm-8"> <div class="title">Biophysical fitness landscape design traps viral evolution</div> <div class="author"> <em>Vaibhav Mohanty</em> and Eugene I. Shakhnovich </div> <div class="periodical"> <em>bioRxiv:2025.03.30.646233</em>, 2025 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1101/2025.03.30.646233" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> </div> <div class="abstract hidden"> <p>Evolutionary adaptation is often visualized as a population’s stochastic climb toward the top of a fitness landscape. While there exist approaches to design or synthetically evolve proteins into desired structures, there is a lack of methodology for designing, tuning, and quantitatively reshaping the fitness landscapes themselves on which protein evolution takes place. Here, we introduce foundational principles of fitness landscape design (FLD) to customize the structural peaks and valleys of biophysical fitness landscapes with quantitative accuracy, offering robust control of long-term evolutionary outcomes. Our FLD algorithms use stochastic optimization of a chemically derived biophysical fitness model to consistently discover optimal antibody ensembles which force a target protein to evolve according to a user-specified target fitness landscape. We then apply FLD to suppress the fitnesses of two SARS-CoV-2 genotype neutral networks and to discover proactive vaccines that preemptively restrict escape variant fitness trajectories before they arise.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <img src="/assets/img/publication_preview/g4.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="g4.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="ross_thermally_2025" class="col-sm-8"> <div class="title">Thermally activated irreversible homogenization of G-quadruplexes in an ALS/FTD-associated nucleotide expansion</div> <div class="author"> Daniel Ross, Olivia Lewis, Olivia McLean, Sundeep Bhanot, Shane Donahue, Rachael Baker, Randi Dias, David Eagerton, <em>Vaibhav Mohanty</em>, and Bidyut K. Mohanty </div> <div class="periodical"> <em>bioRxiv:2025.06.02.657482</em>, 2025 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1101/2025.06.02.657482" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> </div> <div class="abstract hidden"> <p>A significant proportion of familial amyotrophic lateral sclerosis (ALS) and frontotemporal dementia (FTD) cases exhibit a substantial copy number expansion of the hexanucleotide GGGGCC/GGCCCC sequence in the C9ORF72 gene. The GGGGCC sequence forms a non-canonical DNA structure called a G-quadruplex (G4) which has been associated with the disease states and with nucleic acid condensate formation, but there remains poor understanding of why the hexanucleotide repeat expansion occurs. An important clue may lie in the thermodynamic properties of various G4 configurations. Here, we use temperature-swept CD spectroscopy to observe configurational homogenization of an initially heterogeneous population of G4s over a small range of temperatures, demonstrating phase transition-like behavior. We further show that this reaction is irreversible, suggesting that the metastable non-parallel G4s are trapped over long timescales by a transition state barrier; upon heating, the relative height of the free energy barrier is lowered, and a thermally activated metastable-to-stable transition to parallel G4s rapidly occurs. We provide an analytical theory based on a two-state thermodynamic model which offers mechanistic insight into the experimental evidence. These findings suggest that kinetic regulation of non-canonical DNA structures may play a role in cellular homeostasis or disease pathogenesis.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <img src="/assets/img/publication_preview/prgp.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="prgp.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="sappington_probabilistic_2025" class="col-sm-8"> <div class="title">Probabilistic genotype-phenotype maps reveal mutational robustness of RNA folding, spin glasses, and quantum circuits</div> <div class="author"> Anna Sappington and <em>Vaibhav Mohanty</em> </div> <div class="periodical"> <em>Physical Review Research</em>, 2025 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1103/PhysRevResearch.7.013118" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> </div> <div class="abstract hidden"> <p>Recent studies of genotype-phenotype maps have reported universally enhanced phenotypic robustness to genotype mutations, a feature essential to evolution. Virtually all of these studies make a simplifying assumption that each genotype—represented as a sequence—maps deterministically to a single phenotype, such as a discrete structure. Here we introduce probabilistic genotype-phenotype (PrGP) maps, where each genotype maps to a vector of phenotype probabilities, as a more realistic and universal language for investigating robustness in a variety of physical, biological, and computational systems. We study three model systems to show that PrGP maps offer a generalized framework which can handle uncertainty emerging from various physical sources: (1) thermal fluctuation in RNA folding, (2) external field disorder in the spin-glass ground state search problem, and (3) superposition and entanglement in quantum circuits, which are realized experimentally on IBM quantum computers. In all three cases, we observe a biphasic robustness scaling which is enhanced relative to random expectation for more frequent phenotypes and approaches random expectation for less frequent phenotypes. We derive an analytical theory for the behavior of PrGP robustness, and we demonstrate that the theory is highly predictive of empirical robustness.</p> </div> </div> </div> </li> </ol> <h2 class="bibliography">2024</h2> <ol class="bibliography"><li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <img src="/assets/img/publication_preview/biophysical.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="biophysical.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="wang_biophysical_2024" class="col-sm-8"> <div class="title">Biophysical principles predict fitness of SARS-CoV-2 variants</div> <div class="author"> Dianzhuo Wang, Marian Huot, <em>Vaibhav Mohanty</em>, and Eugene I. Shakhnovich </div> <div class="periodical"> <em>Proceedings of the National Academy of Sciences</em>, 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1073/pnas.2314518121" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> </div> <div class="abstract hidden"> <p>SARS-CoV-2 employs its spike protein’s receptor binding domain (RBD) to enter host cells. The RBD is constantly subjected to immune responses, while requiring efficient binding to host cell receptors for successful infection. However, our understanding of how RBD’s biophysical properties contribute to SARS-CoV-2’s epidemiological fitness remains largely incomplete. Through a comprehensive approach, comprising large-scale sequence analysis of SARS-CoV-2 variants and the identification of a fitness function based on binding thermodynamics, we unravel the relationship between the biophysical properties of RBD variants and their contribution to viral fitness. We developed a biophysical model that uses statistical mechanics to map the molecular phenotype space, characterized by dissociation constants of RBD to ACE2, LY-CoV016, LY-CoV555, REGN10987, and S309, onto an epistatic fitness landscape. We validate our findings through experimentally measured and machine learning (ML) estimated binding affinities, coupled with infectivity data derived from population-level sequencing. Our analysis reveals that this model effectively predicts the fitness of novel RBD variants and can account for the epistatic interactions among mutations, including explaining the later reversal of Q493R. Our study sheds light on the impact of specific mutations on viral fitness and delivers a tool for predicting the future epidemiological trajectory of previously unseen or emerging low-frequency variants. These insights offer not only greater understanding of viral evolution but also potentially aid in guiding public health decisions in the battle against COVID-19 and future pandemics.</p> </div> </div> </div> </li></ol> <h2 class="bibliography">2023</h2> <ol class="bibliography"> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <img src="/assets/img/publication_preview/robound.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="robound.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="mohanty_maximum_2023" class="col-sm-8"> <div class="title">Maximum mutational robustness in genotype–phenotype maps follows a self-similar blancmange-like curve</div> <div class="author"> <em>Vaibhav Mohanty</em>, Sam F. Greenbury, Tasmin Sarkany, Shyam Narayanan, Kamaludin Dingle, Sebastian E. Ahnert, and Ard A. Louis </div> <div class="periodical"> <em>Journal of The Royal Society Interface</em>, 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1098/rsif.2023.0169" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> </div> <div class="abstract hidden"> <p>Phenotype robustness, defined as the average mutational robustness of all the genotypes that map to a given phenotype, plays a key role in facilitating neutral exploration of novel phenotypic variation by an evolving population. By applying results from coding theory, we prove that the maximum phenotype robustness occurs when genotypes are organized as bricklayer’s graphs, so-called because they resemble the way in which a bricklayer would fill in a Hamming graph. The value of the maximal robustness is given by a fractal continuous everywhere but differentiable nowhere sums-of-digits function from number theory. Interestingly, genotype–phenotype maps for RNA secondary structure and the hydrophobic-polar (HP) model for protein folding can exhibit phenotype robustness that exactly attains this upper bound. By exploiting properties of the sums-of-digits function, we prove a lower bound on the deviation of the maximum robustness of phenotypes with multiple neutral components from the bricklayer’s graph bound, and show that RNA secondary structure phenotypes obey this bound. Finally, we show how robustness changes when phenotypes are coarse-grained and derive a formula and associated bounds for the transition probabilities between such phenotypes.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <img src="/assets/img/publication_preview/spinglass.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="spinglass.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="mohanty_robustness_2023" class="col-sm-8"> <div class="title">Robustness and stability of spin-glass ground states to perturbed interactions</div> <div class="author"> <em>Vaibhav Mohanty</em> and Ard A. Louis </div> <div class="periodical"> <em>Physical Review E</em>, 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1103/PhysRevE.107.014126" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> </div> <div class="abstract hidden"> <p>Across many problems in science and engineering, it is important to consider how much the output of a given system changes due to perturbations of the input. Here, we investigate the glassy phase of ±J spin glasses at zero temperature by calculating the robustness of the ground states to flips in the sign of single interactions. For random graphs and the Sherrington-Kirkpatrick model, we find relatively large sets of bond configurations that generate the same ground state. These sets can themselves be analyzed as subgraphs of the interaction domain, and we compute many of their topological properties. In particular, we find that the robustness, equivalent to the average degree, of these subgraphs is much higher than one would expect from a random model. Most notably, it scales in the same logarithmic way with the size of the subgraph as has been found in genotype-phenotype maps for RNA secondary structure folding, protein quaternary structure, gene regulatory networks, as well as for models for genetic programming. The similarity between these disparate systems suggests that this scaling may have a more universal origin.</p> </div> </div> </div> </li> </ol> <h2 class="bibliography">2020</h2> <ol class="bibliography"><li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <img src="/assets/img/publication_preview/5Dtonnetz.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="5Dtonnetz.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="mohanty_5-dimensional_2020" class="col-sm-8"> <div class="title">A 5-dimensional Tonnetz for nearly symmetric hexachords</div> <div class="author"> <em>Vaibhav Mohanty</em> </div> <div class="periodical"> <em>Journal of Mathematics and Music</em>, 2020 </div> <div class="periodical"> </div> <div class="links"> <a href="https://doi.org/10.1080/17459737.2020.1799087" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> </div> </div> </div> </li></ol> <h2 class="bibliography">2019</h2> <ol class="bibliography"><li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <img src="/assets/img/publication_preview/graphene.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="graphene.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="mohanty_lazy_2019" class="col-sm-8"> <div class="title">Lazy electrons in graphene</div> <div class="author"> <em>Vaibhav Mohanty</em> and Eric J. Heller </div> <div class="periodical"> <em>Proceedings of the National Academy of Sciences</em>, Sep 2019 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1073/pnas.1908624116" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> </div> <div class="abstract hidden"> <p>Within a tight-binding approximation, we numerically determine the time evolution of graphene electronic states in the presence of classically vibrating nuclei. There is no reliance on the Born–Oppenheimer approximation within the p-orbital tight-binding basis, although our approximation is “atomically adiabatic”: the basis p-orbitals are taken to follow nuclear positions. Our calculations show that the strict adiabatic Born–Oppenheimer approximation fails badly. We find that a diabatic (lazy electrons responding weakly to nuclear distortions) Born–Oppenheimer model provides a much more accurate picture and suggests a generalized many-body Bloch orbital-nuclear basis set for describing electron–phonon interactions in graphene.</p> </div> </div> </div> </li></ol> <h2 class="bibliography">2018</h2> <ol class="bibliography"> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <img src="/assets/img/publication_preview/dodecatonic.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="dodecatonic.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="mohanty_dodecatonic_2018" class="col-sm-8"> <div class="title">Dodecatonic Cycles and Parsimonious Voice-Leading in the Mystic-Wozzeck Genus</div> <div class="author"> <em>Vaibhav Mohanty</em> </div> <div class="periodical"> <em>arXiv:1805.11087</em>, Sep 2018 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> </div> <div class="abstract hidden"> <p>This paper develops a unified voice-leading model for the genus of mystic and Wozzeck chords. These voice-leading regions are constructed by perturbing symmetric partitions of the octave, and new Neo-Riemannian transformations between nearly symmetric hexachords are defined. The behaviors of these transformations are shown within visual representations of the voice-leading regions for the mystic-Wozzeck genus.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <img src="/assets/img/publication_preview/dki.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="dki.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="mohanty_comparison_2018" class="col-sm-8"> <div class="title">Comparison of cumulant expansion and q-space imaging estimates for diffusional kurtosis in brain</div> <div class="author"> <em>Vaibhav Mohanty</em>, Emilie T. McKinnon, Joseph A. Helpern, and Jens H. Jensen </div> <div class="periodical"> <em>Magnetic Resonance Imaging</em>, Sep 2018 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1016/j.mri.2017.12.030" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> </div> <div class="abstract hidden"> <p>Purpose To compare estimates for the diffusional kurtosis in brain as obtained from a cumulant expansion (CE) of the diffusion MRI (dMRI) signal and from q-space (QS) imaging. Theory and methods For the CE estimates of the kurtosis, the CE was truncated to quadratic order in the b-value and fit to the dMRI signal for b-values from 0 up to 2000s/mm2. For the QS estimates, b-values ranging from 0 up to 10,000s/mm2 were used to determine the diffusion displacement probability density function (dPDF) via Stejskal’s formula. The kurtosis was then calculated directly from the second and fourth order moments of the dPDF. These two approximations were studied for in vivo human data obtained on a 3T MRI scanner using three orthogonal diffusion encoding directions. Results The whole brain mean values for the CE and QS kurtosis estimates differed by 16% or less in each of the considered diffusion encoding directions, and the Pearson correlation coefficients all exceeded 0.85. Nonetheless, there were large discrepancies in many voxels, particularly those with either very high or very low kurtoses relative to the mean values. Conclusion Estimates of the diffusional kurtosis in brain obtained using CE and QS approximations are strongly correlated, suggesting that they encode similar information. However, for the choice of b-values employed here, there may be substantial differences, depending on the properties of the diffusion microenvironment in each voxel.</p> </div> </div> </div> </li> </ol> </div> </article> </div> </div> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="/assets/js/bootstrap.bundle.min.js"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@5.0.0/imagesloaded.pkgd.min.js" integrity="sha256-htrLFfZJ6v5udOG+3kNLINIKh2gvoKqwEhHYfTTMICc=" crossorigin="anonymous"></script> <script defer src="/assets/js/masonry.js?a0db7e5d5c70cc3252b3138b0c91dcaf" type="text/javascript"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.1.0/dist/medium-zoom.min.js" integrity="sha256-ZgMyDAIYDYGxbcpJcfUnYwNevG/xi9OHKaR/8GK+jWc=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js?85ddb88934d28b74e78031fd54cf8308"></script> <script src="/assets/js/no_defer.js?2781658a0a2b13ed609542042a859126"></script> <script defer src="/assets/js/common.js?e0514a05c5c95ac1a93a8dfd5249b92e"></script> <script defer src="/assets/js/copy_code.js?c8a01c11a92744d44b093fc3bda915df" type="text/javascript"></script> <script defer src="/assets/js/jupyter_new_tab.js?d9f17b6adc2311cbabd747f4538bb15f"></script> <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script> <script async src="https://badge.dimensions.ai/badge.js"></script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.js" integrity="sha256-MASABpB4tYktI2Oitl4t+78w/lyA+D7b/s9GEP0JOGI=" crossorigin="anonymous"></script> <script src="/assets/js/mathjax-setup.js?a5bb4e6a542c546dd929b24b8b236dfd"></script> <script defer src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6" crossorigin="anonymous"></script> <script defer src="/assets/js/progress-bar.js?2f30e0e6801ea8f5036fa66e1ab0a71a" type="text/javascript"></script> <script src="/assets/js/vanilla-back-to-top.min.js?f40d453793ff4f64e238e420181a1d17"></script> <script>
    addBackToTop();
  </script> </body> </html>